AuthenticationAndAuthorization/MigrationApp/DataLoaderService.cs

```diff
@@ -0,0 +1,110 @@
+﻿using DataAccounts;
+using DataAccounts.Entitys.MovieEntitys;
+using DataAccounts.Entitys;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+using DataAccounts.Repositories;
+using System.Data;
+
+namespace DataLoader
+{
+    public class DataLoaderService
+    {
+        private readonly string _jsonData = Path.Combine(Directory.GetCurrentDirectory(),"data.json");
+        private readonly ApplicationDbContext _context;
+        private readonly IUserRepository _userRepositories;
+
+        public DataLoaderService(ApplicationDbContext context, IUserRepository userRepositories)
+        {
+            _context = context;
+            _userRepositories = userRepositories;
+        }
+
+        public async Task LoadDataAsync()
+        {
+            var jsonData = File.ReadAllText(_jsonData);
+            var rootJson = JsonConvert.DeserializeObject<RootJson>(jsonData);
+
+            await LoadUsersAsync(rootJson.Users);
+            await LoadGenresAsync(rootJson.Genres);
+            await LoadMoviesAsync(rootJson.Movies);
+        }
+
+        public async Task LoadUsersAsync(List<User> users)
+        {
+            foreach (var userJson in users)
+            {
+                try
+                {
+                    string role = UserConstants.User;
+
+                    var user = new User
+                    {
+                        UserName = userJson.UserName,
+                        Email = userJson.Email,
+                        ExternalId = userJson.Id.ToString(),
+                    };
+
+                    if (user.Email.Equals("admin@example.com", StringComparison.OrdinalIgnoreCase))
+                    {
+                        role = UserConstants.Admin;
+                    }
+
+                    await _userRepositories.Register(user, userJson.PasswordHash, role);
```

Comment:
Does it work with hash? Can you login in this user account? I assume it will not work with hash and you will have to create another method to insert user. Because it doesn't seem to be a registration, but rather import/migration, and their hash might not work with your system.

I propose to register them with some default password and each user will change his/her password when logged in